routing/payment_lifecycle.go

```diff
@@ -17,7 +17,7 @@ import (
 type errNoRoute struct {
 	// lastError is the error encountered during the last payment attempt,
 	// if at least one attempt has been made.
-	lastError *htlcswitch.ForwardingError
+	lastError error
```

Comment:
Why should concrete type be abandoned here? We'll just need to type assert it elsewhere to uncover it again.

Reply:
Because it can also be `ErrUnreadableFailureMessage` which is not of type `ForwardingError`. That explicit type assert might be good, to force users to not forget this case.